lib/match/runner.rb

```diff
@@ -90,5 +96,28 @@ def profile(params: nil, certificate_id: nil)
 
       return uuid
     end
+
+    def device_count_different?(params: nil, profile: nil)
```

Comment:
By moving out the `force_for_new_devices` check and getting rid of the type detection.